public/app/features/alerting/unified/components/expressions/util.ts

```diff
@@ -40,29 +42,65 @@ const formatLabels = (labels: Labels): string => {
     .join(', ');
 };
 
+interface DecodedNamespace {
+  name: string;
+  parents: string[];
+}
+
 /**
  * After https://github.com/grafana/grafana/pull/74600,
  * Grafana folder names will be returned from the API as a combination of the folder name and parent UID in a format of JSON array,
  * where first element is parent UID and the second element is Title.
+ *
+ * Here we parse this to return the name of the last folder and the array of parent folders
  */
-const decodeGrafanaNamespace = (namespace: CombinedRuleNamespace): string => {
+const decodeGrafanaNamespace = (namespace: CombinedRuleNamespace): DecodedNamespace => {
   const namespaceName = namespace.name;
 
   if (isCloudRulesSource(namespace.rulesSource)) {
-    return namespaceName;
+    return {
+      name: namespaceName,
+      parents: [],
+    };
   }
 
   try {
-    return JSON.parse(namespaceName).at(-1) ?? namespaceName;
+    const folderParts: string[] = JSON.parse(namespaceName);
+    if (!Array.isArray(folderParts)) {
+      throw new Error('Failed to parse Grafana folder');
```

Comment:
why do we throw an error in here? is not possible to have only the namespace (no array)?

Reply:
Certainly! In that case it will fall through to the `catch` handler and just return the name of the folder with an empty array of parents :)